pkg/cmd/server/admission/init.go

```diff
@@ -22,6 +23,7 @@ type PluginInitializer struct {
 	RESTClientConfig      restclient.Config
 	Informers             shared.InformerFactory
 	ClusterQuotaMapper    clusterquotamapping.ClusterQuotaMapper
+	DefaultRegistryFn     imageapi.DefaultRegistryFunc
```

Comment:
Didn't know this already existed.  It's wrong.  There are N aliases for the registry and the admission plugin must recognize all of them.  There should be a separate function for "get me the default name to use for making references in the cluster", but there are definitely cases where you need all the aliases and those aliases can change.